### PR TITLE
Start up a troubleshooting guide specifically for Ray Serve

### PR DESCRIPTION
**Pull request by chelseatroy**
### Goal(s) of this Pull Request:

- Add a troubleshooting guide for ray serve with the text of what we know from Fakespot so far. 
- 
This is not a complete troubleshooting guide, nor is it intended to be. It's a starting point for us to deposit common problems and their fixes as we continue assisting teams with the tool.

### CODE REVIEW

File: examples/image_classifier/README.md [hunk 1]
```diff
@@ -37,4 +37,4 @@
     ```
 
 > [!IMPORTANT]
-> Please refer to the [Troubleshooting](../../README.troubleshooting.md) guide for the known issues and how to resolve them.
+> Please refer to the [Troubleshooting](../../README.troubleshooting.orchestration) guide for the known issues and how to resolve them.
```
Comment (abhi-agg) [inline examples/image_classifier/README.md:40]: @chelseatroy I know this PR is not ready for review but I am just leaving a note now so that I don't forget about it later that the `Important` part of the [section](https://github.com/mozilla/mozmlops/blob/main/src/mozmlops/templates/README.md#using-docker-with-model-orchestration-flows) will also need a similar update 👍🏾

Comment (chelseatroy) [reply examples/image_classifier/README.md:40]: Change made!

File: README.troubleshooting.inference.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+## If you see a redis cleanup pod hanging there in the cluster for your deployment... 
+
+it means some redis error has happened during killing your pod. In this case, do:
+
+    kubectl get raycluster -n <xxx> to get the raycluster name for your deployment.
+
+    kubectl edit rayclusters.ray.io -n <xxx> <ray_cluster_name> to bring up the yaml for the deployment.
+
+    Find finalizers: - ray.io/gcs-ft-redis-cleanup-finalizer and delete those 2 lines and save.
```
Comment (abhi-agg) [inline README.troubleshooting.inference.md:10]: As these steps will only work _after_ a person is able to access the Kubernetes cluster via bastion host, I believe we should just link the [documentation](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/785514640/Deploy+inference+servers+to+production+GKE+using+Ray+Serve#Running-kubectl%2Fhelm-commands) for doing that here.

File: README.troubleshooting.inference.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    This should apply the change and allow kuberay to clean up the ray cluster without needing the redis cleanup thing to run.
+
+Likely this can be due to your pod name being too long, which translates to your application name being too long. SHORTEN your application name.
+Test Deployment Changes From Local
```
Comment (abhi-agg) [inline README.troubleshooting.inference.md:15]: I think `Test Deployment Changes From Local` should be a heading as well. It is independent of the first one.

**Review by abhi-agg — COMMENTED** [review]
Hey @chelseatroy This overall looks good. Just left two comments after which it should be ready to be merged.

Something which we should definitely address in a separate PR is following:

Currently, there are 2 places in our existing documentation where a troubleshooting guide already exists (see [this](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/785514640/Deploy+inference+servers+to+production+GKE+using+Ray+Serve#Troubleshooting) and [this](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/668663821/Ray+Cluster+on+Kubernetes#KubeRay-Operator)). I apologise as I forgot to raise this earlier.

This PR adds one more Troubleshooting guide which will reside in mozmlops.

TL;DR: We should consolidate all of them at one place (either confluence or mozmlops) to save ourselves a maintenance headache and avoid any confusion for our clients.

**Comment by chelseatroy** [comment]
> Hey @chelseatroy This overall looks good. Just left two comments after which it should be ready to be merged.
> 
> Something which we should definitely address in a separate PR is following:
> 
> Currently, there are 2 places in our existing documentation where a troubleshooting guide already exists (see [this](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/785514640/Deploy+inference+servers+to+production+GKE+using+Ray+Serve#Troubleshooting) and [this](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/668663821/Ray+Cluster+on+Kubernetes#KubeRay-Operator)). I apologise as I forgot to raise this earlier.
> 
> This PR adds one more Troubleshooting guide which will reside in mozmlops.
> 
> TL;DR: We should consolidate all of them at one place (either confluence or mozmlops) to save ourselves a maintenance headache and avoid any confusion for our clients.

@abhi-agg changes made from above, review re-requested. 

Thank you for noting these different documentation locations! I took a look at the confluence locations. I think the first one is solving a different problem than this is, despite having the same title. The "Troubleshooting" in that doc is "here's some general tips for developing your deployment," whereas this document is "the specific error messages of specific things that we have seen go wrong and how to fix them." I think we can keep both, but have the titles indicate their different use cases, so I updated the one in Confluence here to "Developing on Your Deployment." 

The _second_ one does appear to be a similar thing to this, but maintained by Brett (is this right?) as opposed to us. I'd like to leave that in the location where Brett is expecting to find it so we can benefit from his updates to that, so I linked that documentation in this troubleshooting README!
